### PR TITLE
Resolve security concern in logging

### PR DESCRIPTION
**Pull request by nyanzebra**
This is to address [security bug in logging](https://msazure.visualstudio.com/One/_workitems/edit/9554724)

According to the bug, we can use the `j` format specifier to reduce to a single-line JSON string. Also, from documentation we can apply this on the outputTemplate itself: https://github.com/serilog/serilog/wiki/Formatting-Output

### CODE REVIEW

**Comment by arsing** [comment]
I thought the plan was to review every log message in case they have any components that might contain newlines, rather than just JSON'ing every log message, no?

**Comment by nyanzebra** [comment]
> 
> 
> I thought the plan was to review every log message in case they have any components that might contain newlines, rather than just JSON'ing every log message, no?

We can certainly do that, though I see the issue being that we circle back to this in the future because we forgot to add a `j` somewhere. If there is some way to enforce this security issue then that is what I want to do :) If it is evaluating each 900 some log messages for the problem that is fine, but I really want to make sure this is not an issue later too... Is there any option you know of to do this? I read through a bit of the docs for SeriLog but maybe there is something more there to do this...

**Comment by arsing** [comment]
I was just thinking it'd make the logs more annoying to read if they were always enclosed in `""` and had JSON escapes, but I guess it's not too big a deal.

File: edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs [hunk 1]
```diff
@@ -46,6 +46,34 @@ LoggerConfiguration TestSinkMap(LoggerSinkConfiguration loggerSinkConfiguration)
             Assert.Equal("3", emittedEvents[0].Properties["Severity"].ToString());
         }
 
+        [Fact]
+        public void NoNewLinesTest()
+        {
+
```
Comment (arsing) [inline edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs:52]:
```
LoggerTest.cs(51,9): error SA1505: An opening brace should not be followed by a blank line [/home/vsts/work/1/s/edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/Microsoft.Azure.Devices.Edge.Util.Test.csproj]
    13 Warning(s)
    1 Error(s)
```

Comment (nyanzebra) [reply edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs]: Ah I see style issue...

**Comment by nyanzebra** [comment]
> 
> 
> I was just thinking it'd make the logs more annoying to read if they were always enclosed in `""` and had JSON escapes, but I guess it's not too big a deal.

I am willing to go a different route in solving this problem :) We can chat about a solution if this seems like the wrong direction

**Review by arsing — DISMISSED** [review]
It would be good to also get a signoff from someone more familiar with the C# code and with dotnet in general.

File: edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs [hunk 2]
```diff
@@ -46,6 +46,33 @@ LoggerConfiguration TestSinkMap(LoggerSinkConfiguration loggerSinkConfiguration)
             Assert.Equal("3", emittedEvents[0].Properties["Severity"].ToString());
         }
 
+        [Fact]
+        public void NoNewLinesTest()
+        {
+            using (var writer = new System.IO.StringWriter())
+            {
+                System.Console.SetOut(writer);
```
Comment (vadim-kovalyov) [inline edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs:54]: nit: import System.
```suggestion
                Console.SetOut(writer);
```

File: edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs [hunk 3]
```diff
@@ -46,6 +46,33 @@ LoggerConfiguration TestSinkMap(LoggerSinkConfiguration loggerSinkConfiguration)
             Assert.Equal("3", emittedEvents[0].Properties["Severity"].ToString());
         }
 
+        [Fact]
+        public void NoNewLinesTest()
+        {
+            using (var writer = new System.IO.StringWriter())
```
Comment (vadim-kovalyov) [inline edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/LoggerTest.cs:52]: nit: import System.IO
```suggestion
            using (var writer = new tringWriter())
```

**Review by vadim-kovalyov — DISMISSED** [review]
LGTM, minor nits

**Review by vadim-kovalyov — APPROVED** [review]

**Pull request merged**